scalagen/src/test/scala/org/scalameta/scalagen/Examples.scala

```diff
@@ -0,0 +1,36 @@
+package org.scalameta.scalagen
+
+import scala.meta._
+import scala.meta.contrib._
+import org.scalameta.scalagen.generators._
+
+/**
+  * Will print the structure of the class as it's toString.
+  * Not super useful, but a simple example to test with
+  */
+class StructuralToString extends ExtensionGenerator("StructuralToString") {
+  override def extend(d: Defn.Class): List[Stat] = {
```

Comment:
To read arguments arguments to the annotation I would include it on some "context.prefix" like macros